spec/system/admin/order_spec.rb

```diff
@@ -339,6 +317,29 @@ def new_order_with_distribution(distributor, order_cycle)
       expect(new_customer.bill_address).to eq(new_customer.ship_address)
 
```

Comment:
Actually, this one fails on line:

```
expect { click_button "Update" }.to change { Customer.count }.by(1)
```

When clicking on Update, this one is called: 

https://github.com/openfoodfoundation/openfoodnetwork/blob/7b2da4dbe44b614aa36c1bc462ed3c340f430a88/app/controllers/spree/admin/orders/customer_details_controller.rb#L20-L35

This particular line: 
```ruby
OrderWorkflow.new(@order).advance_to_payment
````
actually fails, because `@order.state` is still equal to `cart` after this. 

So, I suspect Customer is not saved after all due to this.

I'd invoke @mkllnk some help on that particular point... 🙏

Reply:
That controller action assumes that we add customer details before going to payment but I guess that's different now. Maybe you need to move/copy that code to the last action before payment.

Also, that action only updates the order. It doesn't even try to create a customer. A customer record is created or updated when the order is actually placed. So this assertion in the spec needs to move down to when the order is actually completed.

Does that help?

Reply:
I've tested, but couldn't find something that solve the issue.

Reply:
I don't think I have any more advice on hand without digging into the code myself. Not sure if we need to shelf this for a bit and put it in my todo list...

Reply:
I'll write something equivalent that I've already written here: https://github.com/openfoodfoundation/openfoodnetwork/pull/9183#issuecomment-1144803818

Would you like to assign yourself this one @mkllnk?

Reply:
> Would you like to assign yourself this one @mkllnk?

Okay, I've assigned myself.